packages/backend/server/src/plugins/copilot/providers/openai.ts

```diff
@@ -245,18 +239,18 @@ export class OpenAIProvider
 
       const [system, msgs] = await chatToGPTMessage(messages);
 
-      const modelInstance = options.webSearch
-        ? this.#instance.responses(model)
-        : this.#instance(model, {
-            structuredOutputs: Boolean(options.jsonMode),
-            user: options.user,
-          });
+      const modelInstance = this.#instance(model, {
+        structuredOutputs: Boolean(options.jsonMode),
+        user: options.user,
+      });
 
       const { fullStream } = streamText({
         model: modelInstance,
         system,
         messages: msgs,
-        tools: this.getToolUse(options),
+        tools: {
+          web_search_preview: openai.tools.webSearchPreview(),
+        },
```

Comment:
not all model support web search, need to keep web search config
and use web search tool without responses api will cause answers that do not require a search to also consume search tokens

Reply:
That makes sense. Now I use the websearch parameters passed in by the front end.